MvvmCross-iOSSupport/MvvmCross.iOS.Support.XamarinSidebar/MvxSidebarPresentationAttribute.cs

```diff
@@ -26,18 +25,25 @@ public class MvxSidebarPresentationAttribute : MvxBasePresentationAttribute
         public readonly bool ShowPanel;
 
         /// <summary>
+        /// Animate the View Controller
+        /// </summary>
+        public readonly bool Animated;
+
+        /// <summary>
         /// Initializes a new instance of the <see cref="MvxSidebarPresentationAttribute"/> class.
         /// </summary>
         /// <param name="panel">The panel.</param>
         /// <param name="hintType">Type of the hint.</param>
         /// <param name="showPanel">if set to <c>true</c> [show panel].</param>
         /// <param name="behaviour">The splitview behaviour value</param>
-        public MvxSidebarPresentationAttribute(MvxPanelEnum panel, MvxPanelHintType hintType, bool showPanel, MvxSplitViewBehaviour behaviour = MvxSplitViewBehaviour.None)
+        /// <param name="animated">if set to <c>true</c> animate the View Controller.</param>
+        public MvxSidebarPresentationAttribute(MvxPanelEnum panel, MvxPanelHintType hintType, bool showPanel, MvxSplitViewBehaviour behaviour = MvxSplitViewBehaviour.None, bool animated = true)
```

Comment:
Related to the Codacy issues:

Should I rather use an overloading constructor pattern or make `Animated` a property? I had chosen to follow the pattern already in place with the `MvxSplitViewBehaviour` for this attribute. Looking at other presentation attributes used in Mvx the patterns used seems to vary between constructor with optional parameters and properties. 

Additionally, some Mvx presenter attributes defaults are referenced via a public static field, I do not understand why it is done this way. My suggestion would be to use constructor parameters for the minimum required values and properties for everything else.

From the [docs.microsoft](https://docs.microsoft.com/en-us/dotnet/standard/attributes/writing-custom-attributes#declaring-constructors)

> You can overload the constructor to accommodate different combinations of values. If you also define a property for your custom attribute class, you can use a combination of named and positional parameters when initializing the attribute. Typically, you define all required parameters as positional and all optional parameters as named. In this case, the attribute cannot be initialized without the required parameter. All other parameters are optional

Reply:
I agree that this gives some confusion and we should clean it up. Can you make an issue for this so we can take this up in the 6.0 release? The static fields are to set default values for all the attributes of that type. That way you don't need to inherit a attribute to be able to set a default value.